### ISSUE TITLE
update gopsutil

### ISSUE THREAD

**Issue by tgross** [hashicorp/nomad#6957]
The panic described in #6954 was caused by a missing length check in `gopsutil` which we've contributed upstream as https://github.com/shirou/gopsutil/pull/822. Once that's been merged and released, we should update `gopsutil`

**Comment by tgross** [comment]
https://github.com/shirou/gopsutil/releases/tag/v2.20.1 was release a couple weeks ago, so we should probably make this bump during the 0.11 window.

**Comment by shishir-a412ed** [comment]
@tgross I tried to vendor `gopsutil` into nomad using go vendoring.

$$ govendor fetch github.com/shirou/gopsutil@v2.20.1

It is updating a bunch of files, however it doesn't update the file which has the fix (`/host/host_linux.go`). Any idea what's going wrong?

````
smahajan@smahajan-VirtualBox:~/go/src/github.com/hashicorp/nomad$ git status
On branch vendor_update
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   vendor/vendor.json

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	vendor/github.com/shirou/gopsutil/Gopkg.lock
	vendor/github.com/shirou/gopsutil/Gopkg.toml
	vendor/github.com/shirou/gopsutil/Makefile
	vendor/github.com/shirou/gopsutil/README.rst
	vendor/github.com/shirou/gopsutil/coverall.sh
	vendor/github.com/shirou/gopsutil/doc.go
	vendor/github.com/shirou/gopsutil/mktypes.sh
	vendor/github.com/shirou/gopsutil/v2migration.sh
	vendor/github.com/shirou/gopsutil/windows_memo.rst
````

**Comment by tgross** [comment]
Just checked and https://github.com/shirou/gopsutil/blob/v2.20.1/host/host_linux.go does have the fix, that's strange. The govendor docs say:

```
# Specify a specific version or revision to fetch
govendor fetch golang.org/x/net/context@a4bbce9fcae005b22ae5443f6af064d80a6f5a55
govendor fetch golang.org/x/net/context@v1   # Get latest v1.*.* tag or branch.
govendor fetch golang.org/x/net/context@=v1  # Get the tag or branch named "v1".
```

So I'm wondering if you want to use the syntax for the specific tag, rather than the "latest" (maybe there's something weird about the way it detects latest that isn't playing nice with the library's go mod setup.)

Something worth checking is to run `make vendorfmt` and then check the `git diff` of the `vendor/vendor.json` file. It might have a clue as to what govendor is doing there.

**Comment by shishir-a412ed** [comment]
@tgross 
Looking at the existing `vendor/vendor.json`. It seems I need to vendor `github.com/shirou/gopsutil/host` and NOT `github.com/shirou/gopsutil`

I am getting the following error when I try:

1) govendor fetch github.com/shirou/gopsutil/host@=v2.20.1
2) govendor fetch github.com/shirou/gopsutil/host@v2.20.1

````
Error: No label found for specified version "=v2.20.1" from github.com/shirou/gopsutil/host::github.com/hashicorp/gopsutil/host@=v2.20.1
Failed to fetch package "github.com/shirou/gopsutil/host"
github.com/kardianos/govendor/context.(*Context).Alter
	/home/smahajan/go/src/github.com/kardianos/govendor/context/modify.go:712
github.com/kardianos/govendor/run.(*runner).Modify
	/home/smahajan/go/src/github.com/kardianos/govendor/run/modify.go:152
github.com/kardianos/govendor/run.(*runner).run
	/home/smahajan/go/src/github.com/kardianos/govendor/run/run.go:98
github.com/kardianos/govendor/run.Run
	/home/smahajan/go/src/github.com/kardianos/govendor/run/run.go:44
main.main
	/home/smahajan/go/src/github.com/kardianos/govendor/main.go:35
runtime.main
	/usr/local/go/src/runtime/proc.go:203
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1357
````

Any idea what's going wrong?
Also I observed that `gopsutil` uses `dep` and HashiCorp is using `govendoring`. Not sure if these two being different dependency management systems is causing a problem.

**Comment by greut** [comment]
Here is my (probably incomplete) understandings. The version of docker links to an old x/sys/windows

The recent gopsutils 2.19.12+, https://github.com/shirou/gopsutil/commit/a61c905252e9fe53ad220c50ad94c9ae926deda9, needs a fresher version of x/sys/windows.

Going anywhere above 2.19.6 requires to bump docker, in order to be able to bump x/sys/windows and use gopsutils 2.20.2+

Been there, done that, https://github.com/hashicorp/nomad/pull/5807 -- didn't go through

**Comment by greut** [comment]
Closed by #7350
